src/download/exchangeDownloader.ts

```diff
@@ -40,57 +45,49 @@ export async function downloadRestApi(
   return response;
 }
 
-export function downloadRestApis(
-  restApi: Array<RestApi>,
+export async function downloadRestApis(
+  restApi: RestApi[],
   destinationFolder: string = DEFAULT_DOWNLOAD_FOLDER
 ): Promise<string> {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const promises: Promise<any>[] = [];
-
-  restApi.forEach((api: RestApi) => {
-    promises.push(downloadRestApi(api, destinationFolder));
-  });
-
-  return Promise.all(promises).then(() => destinationFolder);
+  const downloads = restApi.map(api => downloadRestApi(api, destinationFolder));
+  await Promise.all(downloads);
+  return destinationFolder;
 }
 
-function mapCategories(categories): Categories {
+function mapCategories(categories: RawCategories[]): Categories {
   const cats: Categories = {};
   categories.forEach(category => {
-    cats[category["key"]] = category["value"];
+    cats[category.key] = category.value;
   });
   return cats;
 }
 
-function getFileByClassifier(files, classifier): FileInfo {
-  let myFile: FileInfo;
-  files.forEach(file => {
-    if (file["classifier"] === classifier) {
-      myFile = {
-        classifier: file["classifier"],
-        packaging: file["packaging"],
-        externalLink: file["externalLink"],
-        createdDate: file["createdDate"],
-        md5: file["md5"],
-        sha1: file["sha1"],
-        mainFile: file["mainFile"]
-      };
-    }
-  });
-  return myFile;
+function getFileByClassifier(files: FileInfo[], classifier: string): FileInfo {
+  const found = files.find(file => file.classifier === classifier);
+  // There are extra properties we don't want (downloadURL, isGenerated), so we
+  // create a new object that excludes them
+  return {
+    classifier: found.classifier,
+    packaging: found.packaging,
+    externalLink: found.externalLink,
+    createdDate: found.createdDate,
+    md5: found.md5,
```

Comment:
Do we need to keep these elements in the original object at this point? would it make more sense to do this instead basically?

```typescript
delete found.downloadURL;
delete found.isGenerated;
return found;
```

?

Reply:
Is there a reason why they are excluded in the first place?

Reply:
Download Url is excluded because it expires after 24 hours so it shouldn't be there. Also as it is a presigned s3 url it contains creds to download it from s3 for 24 hours and I didn'nt want to save that in git as bots will pick it up. I forget why isGenerated was ecluded